dataflow/src/main/java/org/checkerframework/dataflow/cfg/CFGBuilder.java

```diff
@@ -4908,7 +4919,15 @@ public Node visitWhileLoop(WhileLoopTree tree, Void p) {
             if (tree.getStatement() != null) {
                 scan(tree.getStatement(), p);
             }
-            extendWithExtendedNode(new UnconditionalJump(conditionStart));
+
+            // If the condition is constant true, for the purpose of improving the dataflow analysis
```

Comment:
I find the performance argument rather weak - it's a constant true expression.
I would add a brief comment to the then and else block, just stating something like:
then: `The condition is constant true so we can directly jump back to the loop entry.`
else: `Otherwise, jump back to evaluate the condition.`